packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/plugin-ledger-connector-iroha.ts

```diff
@@ -0,0 +1,684 @@
+} from "@hyperledger/cactus-core";
+
+import {
+  Checks,
+  Logger,
+  LoggerProvider,
+  LogLevelDesc,
+  Http405NotAllowedError,
+} from "@hyperledger/cactus-common";
+
+import {
+  IrohaCommand,
+  IrohaQuery,
+  // InvokeContractV1Request,
+  // InvokeContractV1Response,
+  RunTransactionRequest,
+  RunTransactionResponse,
+} from "./generated/openapi/typescript-axios";
+
+import { RunTransactionEndpoint } from "./web-services/run-transaction-endpoint";
+//import { isWeb3SigningCredentialNone } from "./model-type-guards";
+import { PrometheusExporter } from "./prometheus-exporter/prometheus-exporter";
+import {
+  GetPrometheusExporterMetricsEndpointV1,
+  IGetPrometheusExporterMetricsEndpointV1Options,
+} from "./web-services/get-prometheus-exporter-metrics-endpoint-v1";
+//import { stringify } from "querystring";
+//import { transcode } from "buffer";
+//import { WatchBlocksV1Endpoint } from "./web-services/watch-blocks-v1-endpoint";
+
+export const E_KEYCHAIN_NOT_FOUND = "cactus.connector.iroha.keychain_not_found";
+
+export interface IPluginLedgerConnectorIrohaOptions
+  extends ICactusPluginOptions {
+  rpcToriiPortHost: string;
+  pluginRegistry: PluginRegistry;
+  prometheusExporter?: PrometheusExporter;
+  logLevel?: LogLevelDesc;
+}
+
+export class PluginLedgerConnectorIroha
+  implements
+    IPluginLedgerConnector<
+      never,
+      never,
+      RunTransactionRequest,
+      RunTransactionResponse
+    >,
+    ICactusPlugin,
+    IPluginWebService {
+  private readonly instanceId: string;
+  public prometheusExporter: PrometheusExporter;
+  private readonly log: Logger;
+  private readonly pluginRegistry: PluginRegistry;
+
+  private endpoints: IWebServiceEndpoint[] | undefined;
+  private httpServer: Server | SecureServer | null = null;
+
+  public static readonly CLASS_NAME = "PluginLedgerConnectorIroha";
+
+  public get className(): string {
+    return PluginLedgerConnectorIroha.CLASS_NAME;
+  }
+
+  constructor(public readonly options: IPluginLedgerConnectorIrohaOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(options, `${fnTag} arg options`);
+    Checks.truthy(
+      options.rpcToriiPortHost,
+      `${fnTag} options.rpcToriiPortHost`,
+    );
+    Checks.truthy(options.pluginRegistry, `${fnTag} options.pluginRegistry`);
+    Checks.truthy(options.instanceId, `${fnTag} options.instanceId`);
+
+    const level = this.options.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.instanceId = options.instanceId;
+    this.pluginRegistry = options.pluginRegistry;
+    this.prometheusExporter =
+      options.prometheusExporter ||
+      new PrometheusExporter({ pollingIntervalInMin: 1 });
+    Checks.truthy(
+      this.prometheusExporter,
+      `${fnTag} options.prometheusExporter`,
+    );
+
+    this.prometheusExporter.startMetricsCollection();
+  }
+  deployContract(): Promise<never> {
```

Comment:
@hxlaf Please add a new line to separate methods from each other

Reply:
A new line is added! Thank you!